### ISSUE TITLE
Integrating gocql into OSS-fuzz

### ISSUE THREAD

**Issue by AdamKorcz** [gocql/gocql#1471]
Dear maintainers of gocql

This is an offer to integrate gocql into Googles [OSS-fuzz](https://github.com/google/oss-fuzz) project. I have tested the existing fuzzer on the OSS-fuzz platform, as since I have all the build scripts available, I will be happy to finish the integration. All I need are the email addresses of the maintainers to whom bug reports should be sent. These email addresses will be publicly available and can be changed anytime.

OSS-fuzz is a project by Google that offers free CPU power to run fuzzers for open source projects. On top of the hardware is a platform that schedules fuzz runs and provides access to an in-depth dashboard with statistics of runs and bugs.

If a bug is found, a link to a detailed report is sent to all maintainers on the contact list, and the bug is subject to a 90 days disclosure policy.

While OSS-fuzz is a free service, it is required by maintainers to fix the bugs, so that the resources spent of fuzzing open source software result in better security for the open source landscape.

ADA Logics is a contributor to security of open source software, and we have integrated dozens of projects into OSS-fuzz some of which are Prometheus, fasthttp, fastjson, grpc-gateway, TiDB, Dragonfly, Vitess, and other large Go projects are already integrated as well like Kubernetes and Coredns.

Kind regards
Adam

**Comment by OleksiienkoMykyta** [comment]
@joao-r-reis sounds great to integrate gocql into Googles OSS-fuzz and provide more fuzzing to it, I think it will help to catch more issues.
Should we add it to the future scope?

**Comment by joao-r-reis** [comment]
I think a wider community discussion is needed before any work is done on it but I'm ok with keeping the issue open. It looks like a low priority item though imo.
